include/zenoh-pico/net/types.h

```diff
@@ -21,7 +21,7 @@
 #include "zenoh-pico/private/collection.h"
 #include "zenoh-pico/private/iobuf.h"
 
-#if (ZENOH_LINUX == 1) || (ZENOH_MACOS == 1)
+#if (ZENOH_LINUX == 1) || (ZENOH_MACOS == 1) || (ESP32 == 1) || (ARDUINO_ARCH_ESP32 == 1)
```

Comment:
It would be preferable to use the same naming convention:
ESP32 -> ZENOH_ESP32
ARDUINO_ARCH_ESP32 -> ZENOH_ARDUINO_ESP32